locator/token_metadata.cc

```diff
@@ -799,8 +807,8 @@ future<> token_metadata_impl::update_topology_change_info(dc_rack_fn& get_dc_rac
     std::sort(begin(all_tokens), end(all_tokens));
 
     auto prev_value = std::move(_topology_change_info);
-    _topology_change_info.emplace(token_metadata(std::move(target_token_metadata)),
-        base_token_metadata ? std::optional(token_metadata(std::move(base_token_metadata))): std::nullopt,
+    _topology_change_info.emplace(make_lw_shared<token_metadata>(std::move(target_token_metadata)),
```

Comment:
Can we get back to passing it by value at the end?

Reply:
Yes, but why? I would rather change all the `token_metadata` usages to pointer for uniformity. It's frustrating to deal with many types for the same concept, I saw a lot of ad-hoc `ptr<->val` conversions during this PR. Maybe we should better create a ticket to clean this up and migrate to ptrs (or values)?

Reply:
I assumed there was a reason for it to have been stored by value, perhaps performance? I don't know. If you see no good reason, we can keep it as ptr I guess.